src/tools/MergeVectorTools.h

```diff
@@ -51,7 +51,13 @@ static void mergeSortedVectors(const C* const* vecs, std::size_t size, std::vect
     /// to allow using a priority_queu, which selects the highest element.
     /// we here (counterintuitively) define < as >
     bool operator< (Entry const& rhs) const { return top() > rhs.top(); }
-    const auto & top() const { return *fwdIt; }
+#ifdef __NVCOMPILER
+#define PLMDAUTO typename C::value_type
+#else
+//I prefer letting gcc doing his work
+#define PLMDAUTO auto
```

Comment:
I would avoid setting a macro `PLMDAUTO` in a .h file which could be included somewhere else. I think this is fine also with gcc:
```
// nvc++ cannot deduce the type correctly
const typename C::value_type & top() const { return *fwdIt; }
```

Also, I recently noticed this also with `icpc` installed on SISSA modules. So maybe it is a widespread bug